src/liboslexec/llvm_gen.cpp

```diff
@@ -3583,6 +3583,19 @@ LLVMGEN (llvm_gen_luminance)
 
 
 
+LLVMGEN (llvm_gen_isconstant)
+{
+    Opcode &op (rop.inst()->ops()[opnum]);
+    ASSERT (op.nargs() == 2);
+    Symbol &Result (*rop.opargsym (op, 0));
+    ASSERT (Result.typespec().is_int());
+    Symbol &A (*rop.opargsym (op, 1));
+    rop.llvm_store_value (rop.ll.constant(A.is_constant() ? 1 : 0), Result);
```

Comment:
I don't think there's a distinction to be made here. It is "constant" if and only if the runtime analysis/optimizer figures out that it's a constant.

Sometimes we figure this out very early, sometimes we may make various optimizations before realizing it. 

Any time we are trying to do a direct constant-fold on the call, we can turn it into a simple "1" if at that point it's known already to be constant. If it's not marked as constant yet, we don't know that it won't be discovered to be constant later, so we can't fold a "0".

When all other optimization is done and it's time to generate LLVM IR, then we generate a 1 or 0 based on whether or not it's known to be constant. It may be that in fact, there is no optimization that could possibly have turned something into a constant that happened after the constfold but before this point, so it will always be 0. But I'm not sure, and it may change in the future, so it seems like the safe (and in no way expensive) approach is to check again rather than assume that it's non-constant at this point.